src/message.lisp

```diff
@@ -49,55 +92,185 @@
+  (when data
+    (setf (slot-value message 'id) (getf data :|message_id|)
+          (slot-value message 'text) (getf data :|text|)
+          (slot-value message 'chat) (make-chat (getf data :|chat|))
+          (slot-value message 'entities) (mapcar (lambda (item)
+                                                   (make-entity message item))
+                                                 (getf data :|entities|))
+          (slot-value message 'raw-data) data
+          (slot-value message 'forward-from-chat) (when (getf data :|forward_from_chat|)
+                                                    (make-chat (getf data :|forward_from_chat|)))
+          (slot-value message 'forward-from) (when (getf data :|forward_from_chat|)
+                                               (make-chat (getf data :|forward_from|)))
+          (slot-value message 'forward-sender-name) (getf data :|forward_sender_name|)
+          (slot-value message 'caption) (getf data :|caption|))))
+
+(defclass temporal ()
+  ((duration
+    :initarg :duration
+    :reader get-duration)))
+
+(defclass spatial ()
+  ((height
+    :initarg :height
+    :reader get-height)
+   (width
+    :initarg :width
+    :reader get-width)))
+
+(defclass unispatial ()
+  ((length
+    :initarg :length
+    :reader get-length)))
+
+(defclass file ()
+  ((file-id
+    :initarg :file-id
+    :reader get-file-id)
+   (file-unique-id
+    :initarg :file-unique-id
+    :reader get-file-unique-id)
+   (file-name
+    :initarg :file-name
+    :reader get-file-name)
+   (file-size
+    :initarg :file-size
+    :reader get-file-size)
+   (mime-type
+    :initarg :mime-type
+    :reader get-mime-type)))
+
+(defclass photo (file spatial) ())
+
+(defclass audio (file temporal)
+  ((performer
+    :initarg :performer
+    :reader get-performer)
+   (title
+    :initarg :title
+    :reader get-title)))
+
+(defclass animation (file temporal spatial) ())
+
+(defclass document (file) ())
+
+(defclass video (file temporal spatial) ())
+
+(defclass video-note (file temporal unispatial) ())
+
+(defclass voice (file temporal) ())
+
+;; TODO: premium_animation, thumb, mask_position
+(defclass sticker (file spatial)
+  ((is-animated
+    :initarg :is-animated
+    :reader get-is-animated)
+   (is-video
+    :initarg :is-video
+    :reader get-is-video)
+   (emoji
+    :initarg :emoji
+    :reader get-emoji)
+   (set-name
+    :initarg :set-name
+    :reader get-set-name)))
+
+(defmethod initialize-instance :after ((file file) &key data &allow-other-keys)
+  (when data
+    (let ((slots (mapcar #'closer-mop:slot-definition-name
+                         (closer-mop:class-slots (class-of file)))))
+      (mapc
+       (lambda (slot underscored)
+         (setf (slot-value file slot) (getf data underscored)))
+       slots
+       (mapcar (lambda (slot)
+                 (kebab:to-snake-case (intern (symbol-name slot) :keyword)))
```

Comment:
Probably it will look simpler as `(alexandria:make-keyword (kebab:to-snake-case slot)`. Also, you can use `alexandria:compose` instead of `lambda`:

```lisp
(mapcar (alexandria:compose #'alexandria:make-keyword
                            #'kebab:to-snake-case)
        slots)
```